src/System.Private.CoreLib/shared/System/ReadOnlySpan.Fast.cs

```diff
@@ -154,6 +154,24 @@ internal ReadOnlySpan(ref T ptr, int length)
 #endif
         }
 
+        public ref readonly T this[Index index]
+        {
+            get
+            {
+                return ref this [index.FromEnd ? _length - index.Value : index.Value];
```

Comment:
This will throw an IndexOutOfRangeException for `new Index(0, true)`.  I assume that's the intended behavior?

Reply:
new Index(o, true_ will not throw but it will when used for indexing the span (or other collections). Yes, this is intended behavior